aten/src/ATen/native/SegmentReduce.cpp

```diff
@@ -0,0 +1,81 @@
+#include <ATen/native/SegmentReduce.h>
+
+#include <ATen/ATen.h>
+
+#include <iostream>
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(segment_reduce_stub);
+
+enum ReductionType { MAX };
+const std::map<std::string, ReductionType> reduce2REDUCE = {
+    {"max", MAX},
+};
+
+Tensor _segment_reduce_cpu(
+    const Tensor& data,
+    std::string reduce,
+    const c10::optional<Tensor>& lengths,
+    const c10::optional<Tensor>& indices,
+    int64_t axis,
+    bool unsafe) {
+  TORCH_CHECK(axis == 0, "Currently only dim=0 is supported!");
```

Comment:
pytorch usually supports wrapping of dimensions, so e.g. 0 and -1 would be valid `axis` argument for a 1d tensor, so before checking `axis` do 
```
axis = maybe_wrap_dim(axis, data.ndim());
```

Reply:
you meant "ndimension", right? SG! updated.